### DISCUSSION TITLE
Docs - The edit button goes to the wrong place

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9854] [General Discussions]
Top right of a documentation page has an "Edit on Github" link.

That link takes you to the Godot docs page on github.

I have been told that is not the correct place to edit the docs as "changing the generated files is not allowed", which makes sense.

This page:

[https://docs.godotengine.org/en/stable/tutorials/best_practices/data_preferences.html](https://docs.godotengine.org/en/stable/tutorials/best_practices/data_preferences.html)

Edit link goes to:

[https://github.com/godotengine/godot-docs/blob/4.2/tutorials/best_practices/data_preferences.rst](https://github.com/godotengine/godot-docs/blob/4.2/tutorials/best_practices/data_preferences.rst)

Please fix to make the link go as close to the thing you can actually edit as possible to allow as many people as possible to contribute to Godot, for many it's their first time using Github.

**Comment by AThousandShips** [comment]
This might not be possible, it might be a limitation of how the readthedocs library works

Edit: there seems to be some confusion here, the "editing the generated files is not allowed" is for the class reference, and those pages *don't* show the "edit on GitHub" link, the question is if we should point the editing of older versions to the latest or not

**Reply by Calinou** [reply]
It might be possible to do some JavaScript trickery to override the **Edit on GitHub** URL to always point to the right branch.

We try to always keep URLs compatible between 4.2 and `latest` so it should be fine to do in the 4.x documentation. For the 3.x documentation, I'd make 3.5 point towards `3.6` when clicking **Edit on GitHub**, as it's where the latest 3.x documentation is maintained.

**Reply by AThousandShips** [reply]
Got it confused with the editing of classes (as the OP mentioned not editing generated files, which is unrelated) the pointing of the source should be as simple as changing the `html_context` in `config.py` as I don't think it is used for generating the content, it points to `4.2` in that version, not sure if changing it won't affect anything else though

**Comment by tetrapod00** [comment]
Since https://github.com/godotengine/godot-docs/pull/10376, **Edit on GitHub** links in `4.3` pages now point to the `master` branch instead of the `4.3` branch. We're still waiting for the next batch of cherrypicks to update `stable`, but I think we can consider this generally resolved.
